src/tools/rbd_mirror/PoolReplayer.cc

```diff
@@ -305,6 +305,13 @@ void PoolReplayer<I>::init(const std::string& site_name) {
     return;
   }
 
+  r = m_remote_rados->cluster_fsid(&m_remote_fsid);
```

Comment:
Nit: I'd move this slightly lower, between `librbd::cls_client::mirror_uuid_get()` and `m_remote_rados->ioctx_create()` calls so that failures related to local pool/operating on `m_local_rados` aren't intermixed with failures related to remote pool/operating on `m_remote_rados`.